src/main/java/com/esri/core/geometry/ogc/OGCConcreteGeometryCollection.java

```diff
@@ -501,4 +538,445 @@ public int hashCode() {
 		
 		return hash;
 	}
+	
+	@Override
+	public double distance(OGCGeometry another) {
+		if (this == another)
+			return isEmpty() ? Double.NaN : 0;
+
+		double minD = Double.NaN;
+		for (int i = 0, n = numGeometries(); i < n; ++i) {
+			double d = geometryN(i).distance(another);
+			if (d < minD || Double.isNaN(minD)) {
+				minD = d;
+				if (minD == 0) {
+					break;
+				}
+			}
+		}
+
+		return minD;
+	}
+	
+	//
+	//Relational operations
+	@Override
+	public boolean disjoint(OGCGeometry another) {
+		if (isEmpty() || another.isEmpty())
+			return true;
+
+		if (this == another)
+			return false;
+
+		//TODO: a simple envelope test
+		
+		OGCConcreteGeometryCollection flattened1 = flatten();
+		if (flattened1.isEmpty())
+			return true;
+		OGCConcreteGeometryCollection otherCol = new OGCConcreteGeometryCollection(another, esriSR);
+		OGCConcreteGeometryCollection flattened2 = otherCol.flatten();
+		if (flattened2.isEmpty())
+			return true;
+		
+		for (int i = 0, n1 = flattened1.numGeometries(); i < n1; ++i) {
+			OGCGeometry g1 = flattened1.geometryN(i);
+			for (int j = 0, n2 = flattened2.numGeometries(); j < n2; ++j) {
+				OGCGeometry g2 = flattened2.geometryN(i);
```

Comment:
This should be geometryN(j) not ..(i).

Reply:
Fixed

Reply:
@stolstov I'm still seeing `OGCGeometry g2 = flattened2.geometryN(i);` using `i` and not `j`.

Reply:
@mbasmanova Sorry about that. I've missed this place. Please, change it locally if you plan testing this, I'll fix this later tonight.

Reply:
@mbasmanova Should be fixed now. I've added a test for this case.